packages/simple-ica/src/callback.rs

```diff
@@ -0,0 +1,45 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{to_binary, Binary, CosmosMsg, StdResult, WasmMsg};
+
+use crate::StdAck;
+
+/// ReceiveIbcResponseMsg should be de/serialized under `Receive()` variant in a ExecuteMsg
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub struct ReceiveIbcResponseMsg {
+    /// The ID chosen by the caller in the `callback_id`
+    pub id: String,
+    pub msg: StdAck,
+}
+
+impl ReceiveIbcResponseMsg {
+    /// serializes the message
+    pub fn into_binary(self) -> StdResult<Binary> {
+        let msg = ReceiverExecuteMsg::ReceiveIbcResponse(self);
+        to_binary(&msg)
+    }
+
+    /// creates a cosmos_msg sending this struct to the named contract
+    pub fn into_cosmos_msg<T: Into<String>, C>(self, contract_addr: T) -> StdResult<CosmosMsg<C>>
+    where
+        C: Clone + std::fmt::Debug + PartialEq + JsonSchema,
+    {
+        let msg = self.into_binary()?;
+        let execute = WasmMsg::Execute {
+            contract_addr: contract_addr.into(),
+            msg,
+            funds: vec![],
+        };
+        Ok(execute.into())
+    }
+}
+
+/// This is just a helper to properly serialize the above message.
+/// The actual receiver should include this variant in the larger ExecuteMsg enum
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+enum ReceiverExecuteMsg {
+    ReceiveIbcResponse(ReceiveIbcResponseMsg),
```

Comment:
I wonder if those should be prefixed by the callback emitter, like `SimpleIcsControllerReceive`. Otherwise it is less clear this name is dictated by an external contract.

Reply:
Good point. What is it receiving?

@JakeHartnell you wrote this, any other ideas on naming?

Reply:
Actually I think simple-ica should export a macros that writes the enum case into the caller's execute enum.

Reply:
My thought behind the naming is that this would eventually live in a `cw-ibc` package, and be a standardized interface for multiple contracts.

Reply:
Okay, a callback for any ack? And the caller would have to multiplex based on which id they entered and which contract called back?

Makes sense. But I would have to generalise the Ack again. 

I would start with an ICA callback and then imagine a more general format

Reply:
Interesting. Then the raw binary success ack data needs to be passed to the caller and only be deserialized there. Not sure, will think about it.

Adding an enum case via an externally defined maco does not seem to be doable in a beautiful way: https://stackoverflow.com/questions/73465525/how-to-use-macros-to-add-an-enum-case

Reply:
I did a rename, we can review this later.

I want to make it specific and not just "any bytes", pushing that to the receiver, but actually parse it in a protocol-specific manner